planet/clients/orders.py

```diff
@@ -94,6 +97,45 @@ def _stats_url(self):
     def _request(self, url, method, data=None, params=None, json=None):
         return Request(url, method=method, data=data, params=params, json=json)
 
+    def _order_factory(self, contents: dict) -> Order:
+        """Convert API response contents to an instance of Order."""
+        links = contents.get("_links", [])
+        results = []
+
+        for res in links.get("results", []):
+            expires_at = res.get("expires_at", None)
+            if expires_at:
+                expires = arrow.get(expires_at).datetime
+            else:
+                expires = None
+
+            results.append(
+                OrderComponent(delivery=res.get("delivery", None),
+                               name=res.get("name", None),
+                               location=res.get("location", None),
+                               expires=expires))
+
+        created_on = contents.get("created_on", None)
+        if created_on:
+            created = arrow.get(created_on).datetime
+        else:
+            created = None
+
+        last_modified = contents.get("last_modified", None)
+        if last_modified:
+            last_modified = arrow.get(last_modified).datetime
+        else:
+            last_modified = None
+
+        return Order(id=contents["id"],
+                     state=contents["state"],
+                     results=results,
+                     created=created,
+                     last_modified=last_modified,
+                     last_message=contents["last_message"],
+                     error_hints=contents["error_hints"],
+                     request=contents)
```

Comment:
`request` is the user-provided order and delivery criteria. Essential, the data that we post to the API when we create an order.